### PR TITLE
Prefer readonlyspan properties over readonly array fields

### PR DESCRIPTION
**Pull request by NewellClark**
While working on [this analyzer issue](https://github.com/dotnet/runtime/issues/33780) I ran the analyzer against dotnet/runtime and found several violations. I've fixed them.

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]
Nice. Thanks.

File: src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateManaged/InflaterManaged.cs [hunk 1]
```diff
@@ -388,11 +388,11 @@ private bool DecodeBlock(out bool end_of_block_code_seen)
                             }
                             else
                             {
-                                if (symbol < 0 || symbol >= s_extraLengthBits.Length)
+                                if (symbol < 0 || symbol >= ExtraLengthBits.Length)
```
Comment (gfoidl) [inline src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateManaged/InflaterManaged.cs:391]:
```suggestion
                                if ((uint)symbol >= ExtraLengthBits.Length)
```

File: src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpResponseStream.Windows.cs [hunk 2]
```diff
@@ -303,15 +303,15 @@ private void DisposeCore()
                 {
                     flags |= Interop.HttpApi.HTTP_FLAGS.HTTP_SEND_RESPONSE_FLAG_DISCONNECT;
                 }
-                fixed (void* pBuffer = &s_chunkTerminator[0])
+                fixed (void* pBuffer = &ChunkTerminator[0])
```
Comment (gfoidl) [inline src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpResponseStream.Windows.cs:306]:
```suggestion
                fixed (void* pBuffer = &MemoryMarshal.GetReference(ChunkTerminator))
```

**Review by gfoidl — COMMENTED** [review]
Low hanging optimizations...
(I see that's not the initial aim of this PR, but as it's quite easy to fix)

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**